faucet/src/tests.rs

```diff
@@ -17,138 +17,644 @@
+
+	if let mock::RuntimeEvent::EncointerFaucet(Event::FaucetCreated(faucet_account, _)) =
+		last_event::<TestRuntime>().unwrap()
+	{
+		return faucet_account
+	} else {
+		panic!("Faucet not found");
+	}
+}
 #[test]
-fn purpose_id_is_stored_correctly() {
+fn faucet_creation_works() {
 	new_test_ext().execute_with(|| {
-		assert!(ReputationCommitmentsPurposeId::<TestRuntime>::exists());
-		assert_eq!(EncointerFaucet::reputation_commitments_purpose_id(), 0,);
-		assert_eq!(
-			EncointerReputationCommitments::purposes(0),
-			DescriptorType::from_str("EncointerFaucet").unwrap(),
+		System::set_block_number(System::block_number() + 1); // this is needed to assert events
+		let alice = AccountId::from(AccountKeyring::Alice);
+		let cid = register_test_community::<TestRuntime>(None, 0.0, 0.0);
+		let cid2 = register_test_community::<TestRuntime>(None, 10.0, 10.0);
+
+		// insert some purposes
+		EncointerReputationCommitments::do_register_purpose(
+			DescriptorType::from_str("Some Description").unwrap(),
+		)
+		.ok();
+		EncointerReputationCommitments::do_register_purpose(
+			DescriptorType::from_str("Some Description 2").unwrap(),
+		)
+		.ok();
+
+		let whitelist_input: WhiteListType = bounded_vec![cid, cid2];
+		Balances::make_free_balance_be(&alice, 114);
+		let faucet_account = new_faucet(
+			RuntimeOrigin::signed(alice.clone()),
+			FaucetNameType::from_str("Some Faucet Name").unwrap(),
+			100,
+			whitelist_input.clone(),
+			10,
 		);
+
+		let faucet = EncointerFaucet::faucets(&faucet_account).unwrap();
+
+		assert_eq!(faucet.name, FaucetNameType::from_str("Some Faucet Name").unwrap());
+		assert_eq!(faucet.purpose_id, 2);
+		assert_eq!(faucet.whitelist, whitelist_input);
+		assert_eq!(faucet.drip_amount, 10);
+		assert_eq!(faucet.creator, alice.clone());
+		assert_eq!(Balances::free_balance(&alice), 1);
+		assert_eq!(Balances::reserved_balance(&alice), 13);
+		assert_eq!(Balances::free_balance(&faucet_account), 100);
 	});
 }
 
 #[test]
-fn set_drip_amount_errs_with_bad_origin() {
+fn faucet_creation_fails_with_insufficient_balance() {
 	new_test_ext().execute_with(|| {
-		assert_dispatch_err(
-			EncointerFaucet::set_drip_amount(
-				RuntimeOrigin::signed(AccountKeyring::Bob.into()),
-				10000u64,
+		System::set_block_number(System::block_number() + 1); // this is needed to assert events
+		let alice = AccountId::from(AccountKeyring::Alice);
+		let cid = register_test_community::<TestRuntime>(None, 0.0, 0.0);
+		let cid2 = register_test_community::<TestRuntime>(None, 10.0, 10.0);
+
+		let whitelist_input: WhiteListType = bounded_vec![cid, cid2];
+		Balances::make_free_balance_be(&alice, 112);
+
+		assert_err!(
+			EncointerFaucet::create_faucet(
+				RuntimeOrigin::signed(alice.clone()),
+				FaucetNameType::from_str("Some Faucet Name").unwrap(),
+				100,
+				whitelist_input.clone(),
+				10
 			),
-			DispatchError::BadOrigin,
+			Error::<TestRuntime>::InsuffiecientBalance
```

Comment:
If my comment above was right, faucet creation fails, but the reserve amount is locked anyway. should be tested here and fixed if  necessary